internal/kms/vault.go

```diff
@@ -192,11 +192,16 @@ func (vc *vaultConnection) initConnection(config map[string]interface{}) error {
 	if errors.Is(err, errConfigOptionInvalid) {
 		return err
 	}
-	vaultAuthNamespace := vaultNamespace // optional, same as vaultNamespace
+	vaultAuthNamespace := ""
 	err = setConfigString(&vaultAuthNamespace, config, "vaultAuthNamespace")
 	if errors.Is(err, errConfigOptionInvalid) {
 		return err
 	}
+	// if the vaultAuthNamespace key is present and value is empty in config, set
```

Comment:
bit confused about this, previously we were setting `vaultDefaultNamespace` as `vaultNamespace` and move it as a `default` value for `vaultAuthNamespace` .  if `vaultAuthNamespace` key is present and value is empty, dont we actually have to return in `errConfigOptionInvalid` saying invalid ?

Reply:
It gets added due to the Marshalling done at https://github.com/ceph/ceph-csi/blob/devel/internal/kms/vault_tokens.go#L136-L139. returning Error causes an issue to the users who are creating the encrypted PVC without `vaultAuthNamespace`. Currently, we don't check for empty values for the configurations it could be a good improvement if we want to do it. May be as a follow-up PR.

Reply:
okay, this makes it crystal clear, good catch !:sweat_smile:

Reply:
@Madhu-1 , add a line about to the commit msg/ pr description please?  
`Marshalling done at https://github.com/ceph/ceph-csi/blob/devel/internal/kms/vault_tokens.go#L136-L139. `

Reply:
Done

Reply:
lets get this in.. thanks !